src/NuGet.Core/NuGet.Protocol/Exceptions/PackageNotFoundProtocolException.cs

```diff
@@ -11,6 +11,7 @@ namespace NuGet.Protocol
     /// <summary>
     /// Thrown when a package cannot be found on a feed.
     /// </summary>
+    [SerializableAttribute]
     public class PackageNotFoundProtocolException : InvalidCacheProtocolException
```

Comment:
This class needs to implement `ISerializable` properly for `PackageIdentity` to round trip.